test/cljc/sieppari/async/core_async_test.cljc

```diff
@@ -0,0 +1,25 @@
+(ns sieppari.async.core-async-test
+  (:require [clojure.test :as test #?(:clj :refer :cljs :refer-macros) [deftest is #?(:cljs async)]]
+            [sieppari.async :as as]
+            [sieppari.async.core-async]
+            [clojure.core.async :as a]))
+
+(deftest async?-test
+  (is (as/async? (a/go "foo"))))
+
+#?(:clj
+   (deftest core-async-continue-clj-promise-test
+     (let [respond (promise)]
+       (as/continue (a/go "foo") (partial deliver respond))
+       (is (= "foo" @respond))))
+   :cljs
+   (deftest core-async-continue-cljs-callback-test
+     (async done
+       (is (as/continue (a/go "foo")
+                        (fn [response]
+                          (is (= "foo" response))
+                          (done)))))))
```

Comment:
This is fine as is for now, but just an idea for future development: we might want to create our own version of cljs.test's async macro to be used on the Clojure side, so with suitable conditional imports we could use the same test code on both sides.